### PR TITLE
Feature pythonz optional and tests update

### LINKED ISSUE #152: Make pythonz optional
pipenv requires pew, and is very happy with it, but our users have no need for pythonz to be installed.

### PR DESCRIPTION
**Pull request by erinxocon**
This PR does two things:

1) It makes pythonz an extra.  If anyone tries to invoke a command from pew that uses pythonz it will `sys.exit` with a message telling them to `pip install pew[pythonz]`.  This resolves #152 and #146!

2) It fixes the travis tests.  The tests before looked like they were passing but they were not actually.  Tox couldn't find the correct versions of python since all tests were set to use 3.4 versions of python.  Tox would gracefully exit giving a message that it couldn't find the right version of python and then, boom, exit code 0.  I've removed tox on travis, in favor of running pytest directly in parallel with all the other versions of python.  This is faster and more accurate.  The windows tests still use tox, as all versions of python are installed on the VM they spin up when you test so tox works well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ sudo: false
...
 python:
   - "2.7"
+  - "3.3"
   - "3.4"
   - "3.5"
   - "3.6"
   - "pypy"
-  - "2.6"
-  - "3.3"
+  # - "2.6"
```
Comment (berdario) [inline .travis.yml]: I'd be fine with dropping Python2.6 support

@jstasiak I'm afraid that you've been the only one that was using it on that version, is that still the case?

**Comment by berdario** [comment]
Regarding

> The tests before looked like they were passing but they were not actually. Tox couldn't find the correct versions of python since all tests were set to use 3.4 versions of python. Tox would gracefully exit giving a message that it couldn't find the right version of python and then, boom, exit code 0. I've removed tox on travis, in favor of running pytest directly in parallel with all the other versions of python. This is faster and more accurate. The windows tests still use tox, as all versions of python are installed on the VM they spin up when you test so tox works well.

It's not quite true that

> tests were set to use 3.4 versions of python

That was just the version picked to install tox on top (which was the most recent version supported by Travis at the time), in fact you can see that in this same year Travis was running tests again 3.3:

https://travis-ci.org/berdario/pew/jobs/195415325

The exit code 0 was intentional, as you can see here:

https://github.com/berdario/pew/commit/d1660e5add95839fd89c33b61e75dbab71a9ce91

I originally tried to add Python3.5, but Travis CI was lagging behind, and I decided to keep it enabled, but ignore it.... My rationale was that as soon as Travis would've enabled Python3.5, tests would've started to run against it

But apparently Travis changed the way that they now provide Python3 versions (instead of installing all the available ones)

(which is actually cleaner and better, though it breaks the expectation that we had of Travis until now :/ )

**Comment by erinxocon** [comment]
@berdario Ahh I see, I'm relatively new to travis so I wasn't aware that they used to have all python versions installed.  There is probably a way to still make that happen, but this way works for now.  Are there any drawbacks doing it this way that you see?

**Comment by berdario** [comment]
>  There is probably a way to still make that happen, but this way works for now. Are there any drawbacks doing it this way that you see?

https://www.fpcomplete.com/blog/2017/03/continuous-integration

> It can be tempting to use every feature and plugin of a CI system to manage builds, but this is usually counterproductive. In general, let the CI system handle the "where and when" of building, but use your own scripts within the repository for the "how". This lets developers use the same scripts locally and makes it easier to switch to a different CI system in the future should that be desirable.

it's not a big deal, and just having a oneliner + versions of Python to test against is just fine... but as a rule of thumb, I quite like the paragraph above :slightly_smiling_face: 

From that point of view, being able to run the exact same tests as they are run on Travis, by leveraging Tox is quite nice (though I'm not a big fan of Tox)

A nicer example, is another small project of mine, in which the .travis.yml is truly minimalistic :)

https://travis-ci.org/berdario/hcobs/builds

https://github.com/berdario/hcobs/blob/master/.travis.yml

File: .travis.yml (hunk 1 above)
Comment (jstasiak) [reply .travis.yml]: @berdario That was three years ago, today I'm of a position that it's fine to drop support for anything older than 3.4 ;)

**Comment by berdario** [comment]
Good!

**Pull request merged**